### DISCUSSION TITLE
Does FASTER Remote support a cluster deployment of multiple faster servers?

### DISCUSSION THREAD

**Discussion by karthik-shekaran** [microsoft/FASTER#774] [Q&A] [answered]
Hello,
Does the FASTER KV Remote Server support a cluster based deployment of multiple servers for fault tolerance - but mainly all hooked up to the same durable storage and writing and reading from the same kv-database-file (or whatever it is internally called and known as)?

Thanks.

**Comment by badrishc** [comment, marked as answer]
This is not supported in FASTER, being a library building block. You will have to build this yourself if it is necessary for your use case.

**Reply by karthik-shekaran** [reply]
Thanks, Badrish, for the confirmation.

**Reply by cmadh** [reply]
I wonder how the Distributed Prefix Recovery - DPR could play a role here? DPR is more about sharding and not about redundancy, right?

**Reply by cmadh** [reply]
Also, a single FasterKVClient-instance currently can only connect to a single FasterServer. As multiple FasterKVClient-instances would do the same Job twice (serialization etc.) wouldn't it make sense to allow the connection to multiple FasterServers?
